models/demos/falcon7b/tt/falcon_model.py

```diff
@@ -98,7 +95,49 @@ def __init__(
 
         self.layernorm_eps = config.layer_norm_epsilon
 
-    def model_preprocessing(self, llm_mode, input_ids, kv_cache_len, num_input_tokens):
+    def _create_prefill_attn_mask_for_sharded_softmax(
+        self,
+        attention_mask,
+        seq_len,
+        device,
+    ) -> List[tt_lib.tensor.Tensor]:
+        if seq_len == 2048:
+            num_slices = 16
+        elif seq_len == 128:
+            num_slices = 1
+        elif seq_len == 1024:
+            num_slices = 4
+        else:
+            raise ValueError("Unsupported seq_len for optimizations")
+
+        attn_masks_per_slice = []
+        attention_mask_starting_index_per_slice = 0
+        slice_length = (self.config.num_attention_heads * seq_len) // num_slices
+        number_of_attention_mask_elements_used_per_slice = slice_length - seq_len * (slice_length // seq_len)
+        for _ in range(num_slices):
+            torch_attn_mask_per_slice = torch.cat(
+                [
+                    attention_mask[:, :, attention_mask_starting_index_per_slice:, :],
+                    attention_mask[:, :, :attention_mask_starting_index_per_slice, :],
+                ],
+                dim=2,
+            )
+
+            tt_attention_slices = torch2tt_tensor(
+                torch_attn_mask_per_slice,
+                device,
+                tt_memory_config=self.model_config["ATTN_MASK_MEMCFG"],
+                tt_dtype=self.model_config["ATTN_MASK_DTYPE"],
+            )
+            attn_masks_per_slice.append(tt_attention_slices)
+
+            attention_mask_starting_index_per_slice = (
+                attention_mask_starting_index_per_slice + number_of_attention_mask_elements_used_per_slice
+            ) % seq_len
+
+        return attn_masks_per_slice
```

Comment:
Can `_create_prefill_attn_mask_for_sharded_softmax` be commonized as much as possible with the near-duplicate version of the function in `model_utils.py` and imported from there?